lib/ansible/modules/windows/win_service_info.ps1

```diff
@@ -0,0 +1,207 @@
+#!powershell
+
+# Copyright: (c) 2020, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+#AnsibleRequires -CSharpUtil Ansible.Service
+
+$spec = @{
+    options = @{
+        name = @{ type = "str" }
+    }
+    supports_check_mode = $true
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$name = $module.Params.name
+
+$module.Result.exists = $false
+$module.Result.services = @(foreach ($rawService in (Get-Service -Name $name -ErrorAction SilentlyContinue)) {
+    try {
+        $service = New-Object -TypeName Ansible.Service.Service -ArgumentList @(
+            $rawService.Name, [Ansible.Service.ServiceRights]'EnumerateDependents, QueryConfig, QueryStatus'
+        )
+    } catch [Ansible.Service.ServiceManagerException] {
+        # ERROR_ACCESS_DENIED, ignore the service and continue on.
+        if ($_.Exception.InnerException -and $_.Exception.InnerException.NativeErrorCode -eq 5) {
+            $module.Warn("Failed to access service '$($rawService.Name) to get more info, ignoring")
+            continue
+        }
+
+        throw
+    }
+    $module.Result.exists = $true
+
+    $controlsAccepted = @($service.ControlsAccepted.ToString() -split ',' | ForEach-Object -Process {
+        switch ($_.Trim()) {
+            Stop { 'stop' }
```

Comment:
Hrm, all this switch fun seems ripe for a PascalToSnakeCase conversion function (and maybe vice versa for win_service)- I only see a couple of exceptional names...

Reply:
Yea we do have a module util that does the camel case which I should probably use instead. There still needs to be some manual mapping for enums values that don't match up with the `win_service` options which is what I was trying to reflect.

Reply:
Looks like they are just camelCase and not PascalCase. I'll look into adding that functionality in the future.